renku/core/commands/dataset.py

```diff
@@ -663,13 +711,90 @@ def update_datasets(
     progress_context=contextlib.nullcontext,
     commit_message=None,
 ):
-    """Update files from a remote Git repo."""
+    """Update files from a remote location."""
+    ignored_datasets = []
+
+    if ((include or exclude) and short_names and any(
+        d.same_as
+        for d in client.datasets.values() if d.short_name in short_names
+    )):
+        raise errors.UsageError(
+            '--include/--exclude is incompatible with datasets created by'
+            ' `renku dataset import`'
+        )
+
+    short_names_provided = bool(short_names)
+
+    # update imported datasets
+    if not include and not exclude:
+        for dataset in client.datasets.values():
+            if (
+                short_names and dataset.short_name not in short_names or
+                not dataset.same_as
+            ):
+                continue
+
+            uri = dataset.same_as.url
+            provider, err = ProviderFactory.from_uri(uri)
+
+            if not provider:
+                continue
+
+            record = provider.find_record(uri, client)
+
+            if record.is_last_version(
+                uri
+            ) and record.version == dataset.version:
+                continue
+
+            uri = record.latest_uri
+
+            # set extract to false if there are any archives present in the
+            # dataset
```

Comment:
This won't work if there were archives within extracted archive. But, we don't have any other way of telling if the archive was extracted in the first place. We can solve this by adding a flag at import time. I don't think it belongs to this story but maybe we should create an issue for that to not forget it.